### PR TITLE
Sampler object

### PR DESCRIPTION
**Pull request by trungtle**
This sample demonstrates an example of using sampler with different texture units.

### CODE REVIEW

**Comment by trungtle** [comment]
@pjcozzi Ready to merge

**Comment by trungtle** [comment]
@pjcozzi ready!

**Comment by pjcozzi** [comment]
> assets/img/wall.png

We wanted to remove this, right?

File: samples/sampler_object.html [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        }
+    </script>
+
+    <script id="fs" type="x-shader/x-fragment">
+        #version 300 es
+        #define FRAG_COLOR_LOCATION 0
+
+        precision highp float;
+        precision highp int;
+        layout(std140, column_major) uniform;
```
Comment (pjcozzi) [inline samples/sampler_object.html]: This isn't needed.

File: samples/sampler_object.html [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        in vec2 st;
+
+        layout(location = FRAG_COLOR_LOCATION) out vec4 color;
+
+        void main()
+        {
+            if (st.y / st.x < 1.0) {
+                color = texture(material.diffuse[0], st);
+            } else {
+                color = texture(material.diffuse[1], st) * .77;
```
Comment (pjcozzi) [inline samples/sampler_object.html]: I suggest using `0.77` (just like there is `0.0`, not `.0`, in the vertex shader), but some folks do just `.77`.  I don't think it is as popular, but it is valid GLSL.

File: samples/sampler_object.html (hunk 1 above)
Comment (trungtle) [inline samples/sampler_object.html]: Removed

File: samples/sampler_object.html [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+        {
+            document.getElementById('info').innerHTML = 'WebGL 2 is not available.  See <a href="https://www.khronos.org/webgl/wiki/Getting_a_WebGL_Implementation">How to get a WebGL 2 implementation</a>';
+            return;
+        }
+
+        // -- Initialize program
+
+        var program = createProgram(gl, getShaderSource('vs'), getShaderSource('fs'));
+
+        var uniformMvpLocation = gl.getUniformLocation(program,"mvp");
```
Comment (pjcozzi) [inline samples/sampler_object.html]: Use `'`, instead of `"`.  It doesn't matter, but the samples should be consistent.

File: samples/utility.js [hunk 4]
```diff
@@ -27,6 +27,7 @@
         console.log(gl.getShaderInfoLog(fshader));
 
         return program;
+
```
Comment (pjcozzi) [inline samples/utility.js]: No need to add the whitespace here.

**Comment by pjcozzi** [comment]
Just those comments.

**Comment by trungtle** [comment]
@pjcozzi updated small tweaks for comments.

**Comment by pjcozzi** [comment]
Do you have ANGLE enabled?  With Canary on my machine, the GLSL struct with a sampler isn't supported yet:

```
utility.js:25 
utility.js:26 
utility.js:27 
utility.js:25 Structures containing samplers not currently supported in D3D.

utility.js:26 
utility.js:27 
sampler_object.html:90 WebGL: INVALID_OPERATION: getUniformLocation: program not linked
sampler_object.html:91 WebGL: INVALID_OPERATION: getUniformLocation: program not linked
sampler_object.html:92 WebGL: INVALID_OPERATION: getUniformLocation: program not linked
sampler_object.html:192 WebGL: INVALID_OPERATION: useProgram: program not valid
sampler_object.html:214 WebGL: INVALID_OPERATION: drawArraysInstanced: no valid shader program in use
```

**Comment by pjcozzi** [comment]
Updated table in README.md.

**Pull request merged**